datastream-server/src/test/java/com/linkedin/datastream/server/assignment/TestLoadBasedPartitionAssignmentStrategy.java

```diff
@@ -0,0 +1,221 @@
+/**
+ *  Copyright 2021 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.server.assignment;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.commons.lang3.StringUtils;
+import org.mockito.Mockito;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.codahale.metrics.MetricRegistry;
+
+import com.linkedin.datastream.common.Datastream;
+import com.linkedin.datastream.common.DatastreamMetadataConstants;
+import com.linkedin.datastream.common.DatastreamRuntimeException;
+import com.linkedin.datastream.common.zk.ZkClient;
+import com.linkedin.datastream.connectors.DummyConnector;
+import com.linkedin.datastream.metrics.DynamicMetricsManager;
+import com.linkedin.datastream.server.ClusterThroughputInfo;
+import com.linkedin.datastream.server.DatastreamGroup;
+import com.linkedin.datastream.server.DatastreamGroupPartitionsMetadata;
+import com.linkedin.datastream.server.DatastreamTask;
+import com.linkedin.datastream.server.DatastreamTaskImpl;
+import com.linkedin.datastream.server.PartitionThroughputInfo;
+import com.linkedin.datastream.server.providers.PartitionThroughputProvider;
+import com.linkedin.datastream.server.zk.KeyBuilder;
+import com.linkedin.datastream.server.zk.ZkAdapter;
+import com.linkedin.datastream.testutil.DatastreamTestUtils;
+import com.linkedin.datastream.testutil.EmbeddedZookeeper;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+
+
+/**
+ * Tests for {@link LoadBasedPartitionAssignmentStrategy}
+ */
+@Test
+public class TestLoadBasedPartitionAssignmentStrategy {
+  private EmbeddedZookeeper _embeddedZookeeper;
+  private ZkClient _zkClient;
+  private String _clusterName;
+
+  @BeforeMethod
+  public void setup() throws IOException {
+    DynamicMetricsManager.createInstance(new MetricRegistry(), "TestStickyPartitionAssignment");
+    _clusterName = "testCluster";
+    _embeddedZookeeper = new EmbeddedZookeeper();
+    String zkConnectionString = _embeddedZookeeper.getConnection();
+    _embeddedZookeeper.startup();
+    _zkClient = new ZkClient(zkConnectionString);
+  }
+
```

Comment:
One test left is to mock `fetchPartitionThroughputInfo` to throw Exception and add the expected behavior in the tests.

Reply:
What do I assert in that case? How do I test that the `super.assignPartitions(...)` was called without exposing the innards of `StickyPartitionAssignmentStrategy` with `@VisibleForTesting`?

Reply:
One thing I can think of is you move the assignor code to another method exposed with VisibleForTesting and ensure that method is not called, but assignPartitions return new assignment

Reply:
@vmaheshw I don't want this test case to impact the code that much. In order to test whether assignment code is called or not I either need to convert the Assigner to an interface (to mock it and then verify call counts), or save some state in the instance (i.e. something like `assignmentPerformed`) and test against that.

Reply:
Why? If you spy the strategy object itself and verify the method call that you extract, it will work.